salt/states/firewalld.py

```diff
@@ -241,3 +255,117 @@ def present(name,
 
     ret['comment'] = '\'{0}\' was configured.'.format(name)
     return ret
+
+
+def bind(name,
+         interfaces=None,
+         sources=None):
+    '''
+    Ensure a zone is bound to specific interfaces or sources.
+
+    .. versionadded:: Boron
+
+    .. note::
+
+        This state does not enforce the existence of the zone. To ensure that
+        the zone exists, use :mod:`firewalld.present <salt.states.firewalld.present>`.
+    '''
+    ret = {'name': name,
+           'result': False,
+           'changes': {},
+           'comment': ''}
+
+    try:
+        zones = __salt__['firewalld.get_zones']()
+    except CommandExecutionError as err:
+        ret['comment'] = 'Error: {0}'.format(err)
+        return ret
+
+    if name not in zones:
+        ret['result'] = False
+        ret['comment'] = 'Zone {0} does not exist'.format(name)
+        return ret
+
+    interfaces = interfaces or []
+    new_interfaces = []
+    old_interfaces = []
+    try:
+        _current_interfaces = __salt__['firewalld.get_interfaces'](name)
+    except CommandExecutionError as err:
+        ret['comment'] = 'Error: {0}'.format(err)
+        return ret
+    for interface in interfaces:
+        if interface not in _current_interfaces:
+            new_interfaces.append(interface)
+            if not __opts__['test']:
+                try:
+                    __salt__['firewalld.add_interface'](name, interface)
+                except CommandExecutionError as err:
+                    ret['comment'] = 'Error: {0}'.format(err)
+                    return ret
+    for interface in _current_interfaces:
+        if interface not in interfaces:
+            old_interfaces.append(interface)
+            if not __opts__['test']:
+                try:
+                    __salt__['firewalld.remove_interface'](name, interface)
+                except CommandExecutionError as err:
+                    ret['comment'] = 'Error: {0}'.format(err)
+                    return ret
+
+
```

Comment:
Please remove this blank line to satisfy the linter. Then we can likely get this merged. Thanks!